base/dates/conversions.jl

```diff
@@ -19,9 +19,20 @@ function unix2datetime(x)
 end
 # Returns unix seconds since 1970-01-01T00:00:00
 datetime2unix(dt::DateTime) = (value(dt) - UNIXEPOCH)/1000.0
+
+_tzoffset = 0
+function settzoffset()
+    t = floor(time())
+    tm = Libc.TmStruct(t)
+    dtutc = unix2datetime(t)
+    dtlocal = DateTime(tm.year+1900,tm.month+1,tm.mday,tm.hour,tm.min,tm.sec)
+    _tzoffset = Second(dtlocal - dtutc).value
+end
+settzoffset() #set it on load
+tzoffset() = _tzoffset
```

Comment:
Cheers. I didn't think I should use const - since it can change if the user changes system timezone and calls `Dates.settzoffset()`. I used the type assertion since it was a non-const global, and I thought that's encouraged for performance when using globals?

What about the type stability issue - is that referring to the addition of the Float `time()` and the `_tzoffset::Int`? Because Second(dtlocal - dtutc).value will always be an `Int`

Also, it was missing a `global` in the setter, that's the main reason it was broken.